x/incentive/keeper/payout_test.go

```diff
@@ -153,179 +154,189 @@ func (suite *KeeperTestSuite) TestPayoutUSDXMintingClaim() {
-// 				suite.Require().True(found)
-// 				err = suite.keeper.AccumulateHardBorrowRewards(runCtx, rewardPeriod)
-// 				suite.Require().NoError(err)
-// 			}
-
-// 			// Fetch pre-claim balances
-// 			ak := suite.app.GetAccountKeeper()
-// 			preClaimAcc := ak.GetAccount(runCtx, suite.addrs[3])
-
-// 			err = suite.keeper.ClaimHardReward(runCtx, suite.addrs[3], tc.args.multiplier)
-// 			if tc.errArgs.expectPass {
-// 				suite.Require().NoError(err)
-
-// 				// Check that user's balance has increased by expected reward amount
-// 				postClaimAcc := ak.GetAccount(suite.ctx, suite.addrs[3])
-// 				suite.Require().Equal(preClaimAcc.GetCoins().Add(tc.args.expectedReward), postClaimAcc.GetCoins())
-
-// 				if tc.args.isPeriodicVestingAccount {
-// 					vacc, ok := postClaimAcc.(*vesting.PeriodicVestingAccount)
-// 					suite.Require().True(ok)
-// 					suite.Require().Equal(tc.args.expectedPeriods, vacc.VestingPeriods)
-// 				}
-
-// 				// Check that the claim's reward amount has been reset to 0
-// 				claim, found := suite.keeper.GetHardLiquidityProviderClaim(runCtx, suite.addrs[3])
-// 				suite.Require().True(found)
-// 				suite.Require().Equal(c("hard", 0), claim.Reward)
-// 			} else {
-// 				suite.Require().Error(err)
-// 				suite.Require().True(strings.Contains(err.Error(), tc.errArgs.contains))
-// 			}
-// 		})
-// 	}
-// }
+func (suite *KeeperTestSuite) TestPayoutHardLiquidityProviderClaim() {
+	type args struct {
+		deposit                  sdk.Coins
+		borrow                   sdk.Coins
+		rewardsPerSecond         sdk.Coin
+		initialTime              time.Time
+		multipliers              types.Multipliers
+		multiplier               types.MultiplierName
+		timeElapsed              int64
+		expectedRewards          sdk.Coins
+		expectedPeriods          vesting.Periods
+		isPeriodicVestingAccount bool
+	}
+	type errArgs struct {
+		expectPass bool
+		contains   string
+	}
+	type test struct {
+		name    string
+		args    args
+		errArgs errArgs
+	}
+	testCases := []test{
+		{
+			"valid 1 day",
+			args{
+				deposit:                  cs(c("bnb", 10000000000)),
+				borrow:                   cs(c("bnb", 5000000000)),
+				rewardsPerSecond:         c("hard", 122354),
+				initialTime:              time.Date(2020, 12, 15, 14, 0, 0, 0, time.UTC),
+				multipliers:              types.Multipliers{types.NewMultiplier(types.MultiplierName("small"), 1, d("0.25")), types.NewMultiplier(types.MultiplierName("large"), 12, d("1.0"))},
+				multiplier:               types.MultiplierName("large"),
+				timeElapsed:              86400,
+				expectedRewards:          cs(c("hard", 21142771200)), // 10571385600 (deposit reward) + 10571385600 (borrow reward)
+				expectedPeriods:          vesting.Periods{vesting.Period{Length: 32918400, Amount: cs(c("hard", 21142771200))}},
+				isPeriodicVestingAccount: true,
+			},
+			errArgs{
+				expectPass: true,
+				contains:   "",
+			},
+		},
+		{
```

Comment:
Add test with multiple coins on the claim, add similar test for USDX minting rewards

Reply:
Updated payout test in https://github.com/Kava-Labs/kava/pull/788/commits/c63ff89829ca9b789374f3999e0a928104020ecd.

What do you mean by 'similar test for USDX minting rewards'? USDXMintingClaims don't use `MultiRewardPeriod`/`MultiRewardIndex` yet. If the suggestion here is to test multiple reward coins within the current set up (e.g. USDXMinting RewardPeriods params contain two RewardPeriods for bnb/xrp that have different reward denoms such as ukava/hard) I'd rather just update USDXMintingClaims to use `MultiRewardPeriods` and then test multiple reward denoms using the existing testing process from TestPayoutHardLiquidityProviderClaim. I can make an issue for it.

Reply:
Ah my bad, forgot we didn't implement Multi rewards for USDX minting yet. Ignore.